### PR TITLE
RC-3.0 Feature - method for manually setting empty states

### PR DESCRIPTION
**Pull request by dalerasrorov-eb**
Added `shouldShowEmptyState` method to `donut` and `stacked-area` APIs to set the empty state manually (even if data is given) to see the empty states as described in https://github.com/eventbrite/britecharts/issues/483

## Description
Empty state will be known using `shouldShowEmptyState`. For the donut chart, it will show an empty slice with no info for the centered text (we can switch to make it show 0% if needed).
*Update:* changed logic in `drawLegend` to show `0%` if `shouldShowEmptyState` is `true`

## Motivation and Context
https://github.com/eventbrite/britecharts/issues/483

## How Has This Been Tested?
Added two new setter/getter, one for each chart.

## Screenshots (if appropriate):
`donut.sholdShowEmptyState(true)`:
screen shot 2018-02-28 at 3 40 20 pm

`stackedArea.sholdShowEmptyState(true)`:
screen shot 2018-02-27 at 5 16 44 pm

## Types of changes

- [ ] Refactor (changes the way we code something without changing its functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the [code style guide](https://github.com/eventbrite/britecharts/blob/master/CODESTYLEGUIDE.md) of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/charts/donut.js [hunk 1]
```diff
@@ -156,7 +157,7 @@ define(function(require) {
                 if (highlightedSliceId) {
                     initHighlightSlice();
                 }
-                if (isEmpty && emptyDataConfig.showEmptySlice) {
+                if ((isEmpty && emptyDataConfig.showEmptySlice) || shouldShowEmptyState) {
```
Comment (Golodhros) [inline src/charts/donut.js]: Actually, I wanted to remove the 'isEmpty' variable altogether.
Also, I think we should get rid of 'emptyDataConfig', keeping only 'emptySliceColor' as a regular accessor.

File: src/charts/donut.js [hunk 2]
```diff
@@ -819,6 +820,23 @@ define(function(require) {
             return this;
         };
 
+        /**
+         * Gets or Sets the shouldShowEmptyState state of the chart
+         * The default value is -60
```
Comment (Golodhros) [inline src/charts/donut.js]: Oops

File: src/charts/stacked-area.js [hunk 3]
```diff
@@ -521,7 +523,10 @@ define(function(require){
          * @private
          */
         function cleanData(originalData) {
-            originalData = originalData.length === 0 ? createFakeData() : originalData;
+            originalData = 
+                originalData.length === 0 || shouldShowEmptyState
```
Comment (Golodhros) [inline src/charts/stacked-area.js]: Let´s get rid of "originalData.length" here, so only if shouldShowEmptyState is true we will show the empty state.

File: src/charts/stacked-area.js [hunk 4]
```diff
@@ -1343,6 +1348,23 @@ define(function(require){
          */
         exports.axisTimeCombinations = axisTimeCombinations;
 
+        /**
+         * Gets or Sets the shouldShowEmptyState state of the chart
+         * The default value is -60
```
Comment (Golodhros) [inline src/charts/stacked-area.js]: Oops

File: src/charts/donut.js (hunk 2 above)
Comment (DalerAsrorov) [reply src/charts/donut.js]: Copy-pasting is hard :)

**Comment by miglesiasEB** [comment]
Merged into https://github.com/eventbrite/britecharts/tree/release/3.0.0-rc1

Thanks a lot!

**Pull request closed without merging**